### ISSUE TITLE
Extension of algorithm

### ISSUE THREAD

**Issue by lcxiha** [yewzijian/RegTR#14]
Hello，Is this algorithm suitable for underwater point cloud map registration? If appropriate, how should the dataset be processed?Is the dataset unlabeled?（For unsupervised learning）

**Comment by yewzijian** [comment]
Hi. I don’t have experience with underwater point clouds so I am unable to
advise. The work is a supervised learning algorithm and requires ground
truth registrations for training.

**Comment by lcxiha** [comment]
Thanks a lot! There are still some questions I would like to consult: 1. The ground truth (label) is the true transformation R and T between two point clouds, right? 2.Is it the same dataset as the one used in the paper predator?
       Looking forward to your reply.

**Comment by yewzijian** [comment]
1. Yes the ground truth label is the pose (rotation+translation). Although
internally it converts them into the ground truth corresponding locations
for training, since that’s what the network outputs.

2. Yes, the dataset is the same as Predator.


Zi Jian

**Comment by lcxiha** [comment]
Ok,thanks.If I want to use this algorithm for unlabeled datasets, how can I obtain labels for ground truth? Do you have any good suggestions? Looking forward to your reply.

**Comment by lcxiha** [comment]
Hi,Does"Although internally it converts them into the ground truth corresponding locations for training, since that’s what the network outputs."refer to Positional encodings?

**Comment by yewzijian** [comment]
> Ok,thanks.If I want to use this algorithm for unlabeled datasets, how can I obtain labels for ground truth? Do you have any good suggestions? Looking forward to your reply.

This is tricky and depends on your problem. Some possible solutions are rely on external sensors, e.g. mocap systems, or semi-manual/manual registrations.

> Hi,Does"Although internally it converts them into the ground truth corresponding locations for training, since that’s what the network outputs."refer to Positional encodings?

What I meant was that the training loss is based on the point coordinates, and not the rotation/translation.

**Comment by yewzijian** [comment]
Lidar Point clouds like those from KITTI should be fine. After the paper
publication, I did try it briefly on KITTI point cloud matching and a
KITTI-trained model worked well.

Not sure what you mean by odometry vs SLAM loop back. But for odometry
where you already have a good initial guess, local algorithms like ICP tend
to work well and give precise results. RegTR will work but might not be the
best use of the algorithm.

**Comment by yewzijian** [comment]
If you have the rotation and translation matrices of your dataset you
should be able train RegTR on your dataset. As mentioned, the algorithm
should work on lidar point clouds like those in KITTI (although some
parameter tuning might necessary).

Zi Jian

**Comment by lcxiha** [comment]
Thank you very much for your patient answer. I have another question: May I ask if the dataset used in the paper is in the global coordinate system or in the carrier coordinate system.Looking forward to your reply.

**Comment by yewzijian** [comment]
Not 100% sure, but I suspect the coordinates of the points are w.r.t. to the camera, i.e. origin is where the camera is.

**Comment by lcxiha** [comment]
Thanks a lot! Given the rotation matrix and translation matrix, I want to visualize the point cloud registration of two frames with overlapping rate. How do I build this code?

**Comment by yewzijian** [comment]
For visualizing cloud clouds, look at my demo code (VTK based). Or you can
use the visualization functions provided in open3d.

**Comment by lcxiha** [comment]
Sorry, I may not have expressed my question clearly. I want to visualize two point clouds with overlapping frames, but registration is not done with the trained model. Can the alignment of two point clouds (with a certain overlap rate) be performed when the transformation matrix of two point clouds is known?

**Comment by yewzijian** [comment]
Sorry I don’t quite understand. If you already have the transformation
matrix between the point clouds, aren’t they already aligned?


Zi Jian

**Comment by lcxiha** [comment]
I'm sorry, I want to verify if the dataset can be aligned under the provided transformation matrix. Now I know! However, I have a question: why does the test dataset (test-3DMatch_info.pkl and test-3DLoMatch_info.pkl) still have a transformation matrix? Isn't it only the training dataset (train_info. pkl) and the validation dataset (vai_info. pkl) that have labels (transformation matrices)?

**Comment by yewzijian** [comment]
The transformation matrices are used for evaluation.


Zi Jian

**Comment by lcxiha** [comment]
I don't know if my understanding is correct: the transformation matrices in the training dataset and validation dataset are used for training and evaluation, while the transformation matrices in the test dataset are only used for evaluation, that is to evaluate the quality of the registration results of the two point clouds.

**Comment by aniket-gupta1** [comment]
@yewzijian Can you share what parameter file you used for training the model on KITTI?

**Comment by yewzijian** [comment]
Sure, will look for it and post it here by the end of the week.

**Comment by yewzijian** [comment]
Hi @aniket-gupta1 , see the following for my Kitti parameters. I didn't really spend much time tuning these parameters though so you might be able to find better ones.
[kitti.yaml.zip](https://github.com/yewzijian/RegTR/files/11994836/kitti.yaml.zip)

**Comment by aniket-gupta1** [comment]
@yewzijian Thankyou!
